controlplane/eks/api/v1beta1/awsmanagedcontrolplane_types.go

```diff
@@ -173,6 +173,10 @@ type AWSManagedControlPlaneSpec struct { //nolint: maligned
 	// +kubebuilder:default=false
 	DisableVPCCNI bool `json:"disableVPCCNI,omitempty"`
 
+	// FilterPrivateSubnets indicates that the Amazon EKS control plane should only use private subnets.
+	// +optional
+	FilterPrivateSubnets bool `json:"filterPrivateSubnets,omitempty"`
```

Comment:
This doesn't need to be added to older api versions. Just the current version (i.e. the newest).